datadog_checks_base/datadog_checks/base/utils/profiling.py

```diff
@@ -0,0 +1,41 @@
+# (C) Datadog, Inc. 2018-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from threading import Lock
+
+
+class Profiling(object):
+    def __init__(self):
+        self._profiler = None
+        self._running = False
+        self._mutex = Lock()
+
+    def start(self):
+        with self._mutex:
+            if not self._running and self._profiler is None:
+                from ddtrace.profiling import Profiler
+
+                self._profiler = Profiler(service="datadog-agent-integrations")
+                self._profiler.start()
+                self._running = True
+            if not self._running and self._profiler is not None:
+                # This branch is exercise if we have previously stopped the profiler
+                if self._profiler.status != "running":
+                    self._profiler.start()
+                    self._running = True
+
+    def stop(self):
+        if not self._running:
+            return
+
+        with self._mutex:
```

Comment:
Why do we need `self._mutex` when checking if `self._running` is `True` but not when checking if it's `False`?

Could you please add an inline comment somewhere saying why mutex is necessary in the first place?
You can probably include also why this needs to be a singleton while you're at it. 

**And** if you really want to convey more clearly that this should not be initialized outside of the module, you could make the class private as well `s/Profiling/_Profiling/`.